classes/Product.php

```diff
@@ -3195,12 +3195,13 @@ public static function getPriceStatic(
 
         $cart_quantity = 0;
         if ((int) $id_cart) {
-            $cache_id = 'Product::getPriceStatic_' . (int) $id_product . '-' . (int) $id_cart;
+            $cache_id = 'Product::getPriceStatic_' . (int) $id_product . '-' . (int) $id_product_attribute . '-' . (int) $id_cart;
             if (!Cache::isStored($cache_id) || ($cart_quantity = Cache::retrieve($cache_id) != (int) $quantity)) {
                 $sql = 'SELECT SUM(`quantity`)
                 FROM `' . _DB_PREFIX_ . 'cart_product`
                 WHERE `id_product` = ' . (int) $id_product . '
-                AND `id_cart` = ' . (int) $id_cart;
+                AND `id_cart` = ' . (int) $id_cart . '
+				AND `id_product_attribute` = ' . (int) $id_product_attribute;
```

Comment:
Watch out, not sure you can request `id_product_attribute` if `$id_product_attribute` is null.
You should add conditionnal `AND`

Reply:
Fixed. Thanks.